docs/apps/build-system.adoc

```diff
@@ -89,33 +89,54 @@ NOTE:  XP apps deployed via file (locally) are presented with a small blue icon
 
 === Development Mode
 
-Enonic XP supports a so-called Development mode.
-When running in this mode, XP will automatically source JavaScript controllers directly from your project folder.
+[WARNING]
+====
+As the name implies, development mode is intended only for development use and should _not_ be used in production. Doing so would cause both security and performance issues.
+====
 
-This may be convenient if you are developing with pure JavaScript and do not depend on build steps in your project.
+Enonic XP can be run in a special mode known as _development mode_. In this mode, XP will try to detect changes to your project source files, including JavaScript controllers and registered XML schemas. It does this by reading these files directly from their source locations instead of from the application `.jar` file.
 
-You may activate dev mode when starting XP as follows:
+Because of the way XP reads source files directly in development mode, it has a few limitations:
+
+- Files that require any sort of compilation (including transpilation) are not detected unless you run a build process on the side. For instance, if you use TypeScript or JavaScript features that XP doesn't support, these files won't be picked up by default. To have XP detect them, run a separate build process on the side, such as either a xref:build-system#_continuous_building[continuous build] or a webpack build.
++
+NOTE: XP uses the  Nashorn JavaScript engine for compiling JavaScript. For an overview over what JavaScript features Nashorn supports, see https://kangax.github.io/compat-table/es6/#nashorn1_8[this feature table].
++
+- XP will not pick up _new_ content types. If you add a new content type after your previous project deploy, XP will not be able to see it. However, changes to _existing_ content types _are_ picked up. XP will pick up new JavaScript controllers as long as it can run them without compilation (as per the previous point).
```

Comment:
> XP will not pick up _new_ content types
Is it true @anatol-sialitski ?

Reply:
I've just test that and as the result a new content type is appeared in Content Studio without redeploy a project. Pay attention that I test that after [this](https://github.com/enonic/xp-gradle-plugin/pull/83) fix in gradle plugin.

Reply:
```suggestion
```